packages/wrangler/src/tail/index.ts

```diff
@@ -201,11 +193,59 @@ export async function tailHandler(args: TailArgs) {
 		logger.log(`Connected to ${scriptDisplayName}, waiting for logs...`);
 	}
 
-	tail.on("close", async () => {
+	const cancelPing = startWebSocketPing();
+	tail.on("close", exit);
+	onExit(exit);
+
+	async function exit() {
+		cancelPing();
 		tail.terminate();
 		await deleteTail();
 		await metrics.sendMetricsEvent("end log stream", {
 			sendMetrics: config.send_metrics,
 		});
-	});
+	}
+
+	/**
+	 * Start pinging the websocket to see if it is still connected.
+	 *
+	 * We need to know if the connection to the tail drops.
+	 * To do this we send a ping message to the backend every few seconds.
+	 * If we don't get a matching pong message back before the next ping is due
+	 * then we have probably lost the connect.
+	 */
+	function startWebSocketPing() {
+		/** The corelation message to send to tail when pinging. */
+		const PING_MESSAGE = Buffer.from("wrangler tail ping");
+		/** How long to wait between pings. */
+		const PING_INTERVAL = 2000;
```

Comment:
This feels a little low - users with poor connections, or those far away from DOs could easily get this timeout, something like 10s would feel a lot better imo

Reply:
Fair comment. I'll bump it. Thanks